File: 2021/day02/p2.pl
#!/usr/bin/env perl
use v5.34;
my ($aim, $pos, $depth) = (0,0,0);
while (<>) {
  if (/forward (\d+)/) { $pos+=$1; $depth += $aim * $1; }
  elsif (/down (\d+)/) { $aim+=$1 }
  elsif (/up (\d+)/) { $aim-=$1 }
}
say $pos * $depth;

File: 2021/day02/p2.py
#!/usr/bin/env python
import fileinput
aim, pos, depth = 0, 0, 0
for line in fileinput.input():
  command, value = line.split()
  match command:
    case 'forward':
      pos += int(value)
      depth += aim * int(value)
    case 'down':
      aim += int(value)
    case 'up':
      aim -= int(value)

print(pos * depth)
